gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -472,48 +450,50 @@ private void setFinalJobState(JobState jobState) {
   }
 
   /**
-   * Commit a finished job.
+   * Check if it is OK to commit the output data of the job.
    */
-  @SuppressWarnings("unchecked")
-  private void commitJob(String jobId, JobState jobState)
-      throws Exception {
-    JobCommitPolicy commitPolicy = JobCommitPolicy.forName(
-        jobState.getProp(ConfigurationKeys.JOB_COMMIT_POLICY_KEY, ConfigurationKeys.DEFAULT_JOB_COMMIT_POLICY));
-
+  private boolean canCommit(JobCommitPolicy commitPolicy, JobState jobState) {
     // Only commit job data if 1) COMMIT_ON_PARTIAL_SUCCESS is used,
-    // or 2) COMMIT_ON_FULL_SUCCESS is used and the job is successful.
-    if (commitPolicy == JobCommitPolicy.COMMIT_ON_PARTIAL_SUCCESS || (
-        commitPolicy == JobCommitPolicy.COMMIT_ON_FULL_SUCCESS
-            && jobState.getState() == JobState.RunningState.SUCCESSFUL)) {
-
-      Class<? extends DataPublisher> dataPublisherClass =
-          (Class<? extends DataPublisher>) Class.forName(jobState.getProp(ConfigurationKeys.DATA_PUBLISHER_TYPE,
-              ConfigurationKeys.DEFAULT_DATA_PUBLISHER_TYPE));
-      Constructor<? extends DataPublisher> dataPublisherConstructor = dataPublisherClass.getConstructor(State.class);
-
-      LOG.info(String.format("Publishing job data of job %s with commit policy %s", jobId, commitPolicy.name()));
+    // or 2) COMMIT_ON_FULL_SUCCESS is used and the job has succeeded.
+    return commitPolicy == JobCommitPolicy.COMMIT_ON_PARTIAL_SUCCESS ||
```

Comment:
In that case, as an editorial comment, COMMIT_ON_PARTIAL_SUCCESS is a bit misleading since we don't check if anything was successful at all. It looks a bit more like "ALWAYS_COMMIT".